Sources/Sentry/SentryEnvelope.m

```diff
@@ -9,10 +11,23 @@ @implementation SentryEnvelopeHeader
 
 // id can be null if no event in the envelope or attachment related to event
 - (instancetype)initWithId:(NSString *_Nullable)eventId
+{
+    SentrySdkInterface *sdkInterface =
+        [[SentrySdkInterface alloc] initWithName:SentryMeta.sdkName
+                                      andVersion:SentryMeta.versionString];
+    self = [self initWithId:eventId andSdkInterface:sdkInterface];
```

Comment:
H: this method always returns an instance of `SentrySdkInterface` but if no name is provided, it should be `null` as @bruno-garcia suggested, a null or empty sdk name even with a version, is just noisy in the json, because its useless

Reply:
In this case, it's not going to be `nil`, because we init `SentrySdkInterface` from `SentryMeta`. If somebody calls 
```
- (instancetype)initWithId:(NSString *_Nullable)eventId
           andSdkInterface:(SentrySdkInterface *_Nullable)sdkInterface
```
with a`sdkInterface` set to `nil` [`SentrySerialization`](https://github.com/getsentry/sentry-cocoa/blob/feat/attach-sdkinfo-to-envelope/Sources/Sentry/SentrySerialization.m#L43-L46) doesn't add it to the JSON.

For me, it doesn't make sense to return `nil` for a init of a data class when some values are empty just to make the JSON smaller. What if I want them to be empty? This would be putting logic about the JSON serialization in a data class. Setting the `SentrySdkInterface` to `nil` on the init of the `SentryEnvelopeHeader` if the values are empty also doesn't make sense to me. What we could do is to not add the `SentrySdkInterface` to the JSON if all its values are empty, but what if I want them to be empty? Furthermore, this is something that shouldn't happen after all. We should always attach the `SentrySdkInterface` to a `SentryEnvelopeHeader`. Optimizing for an edge case by making some values `nil` doesn't make sense for me.

Reply:
to be honest I don't care where `SentrySdkInterface` would be set to null, I do care if we are adding a SentrySdkInterface object to the json that won't help at all, eg:

SentrySdkInterface without a name
SentrySdkInterface with empty or null name and version

https://develop.sentry.dev/sdk/event-payloads/sdk/#attributes
both fields are required, so you either set a valid `sdk` object with name and version or you dont set a `sdk` object or you might break the parsing on the server

Reply:
The recommended init of `SentryEnvelopeHeader` is `initWithId:eventId` that sets the `SdkInfo` for you. If you use `initWithId:eventId:sdkInfo` then you better should know what you are doing. If you create one with name and version empty then it's your fault IMO. I tried it and the server still accepts the envelope.